mlir/lib/Quantum/Transforms/ChainedSelfInversePatterns.cpp

```diff
@@ -30,24 +29,33 @@ namespace {
 struct ChainedHadamardOpRewritePattern : public mlir::OpRewritePattern<CustomOp> {
     using mlir::OpRewritePattern<CustomOp>::OpRewritePattern;
 
-    /// We check if the operation and it's parent are hadamard operations. If so, replace op
-    /// with it's "grandparent".
+    /// We simplify consecutive Hermitian quantum gates and simplifies them.
+    /// Hermitian gates are self-inverse and applying the same gate twice in succession
+    /// cancels out the effect. This pattern rewrites such redundant operations by
+    /// replacing the operation with its "grandparent" operation in the quantum circuit.
     mlir::LogicalResult matchAndRewrite(CustomOp op, mlir::PatternRewriter &rewriter) const override
     {
-        LLVM_DEBUG(dbgs() << "Simplifying the following hadamard operation:\n" << op << "\n");
-        if (op.getGateName().str() != "Hadamard")
-            return failure();
+        LLVM_DEBUG(dbgs() << "Simplifying the following operation:\n" << op << "\n");
 
-        ValueRange qbs = op.getInQubits();
-        auto parentHadamard = dyn_cast<CustomOp>(qbs[0].getDefiningOp());
+        mlir::StringSet<> HermitianOps = {"Hadamard", "PauliX", "PauliY", "PauliZ", "CNOT",
+                                          "CY",       "CZ",     "SWAP",   "Toffoli"};
 
-        if (parentHadamard == nullptr)
+        StringRef OpGateName = op.getGateName();
+        if (HermitianOps.find(OpGateName) == HermitianOps.end())
```

Comment:
`StringSet::contains`

https://llvm.org/doxygen/classllvm_1_1StringSet.html#aa3137f5ea0c0700166fbd5281d11396e

Reply:
Thanks. will fix that. It improves readability